packages/xo-web/src/xo-app/hub/resource-form.js

```diff
@@ -3,38 +3,45 @@ import _ from 'intl'
 import decorate from 'apply-decorators'
 import Icon from 'icon'
 import React from 'react'
+import SingleLineRow from 'single-line-row'
 import Tooltip from 'tooltip'
-import { Container } from 'grid'
-import { SelectPool } from 'select-objects'
-import { error } from 'notification'
+import { Container, Col } from 'grid'
+import { differenceBy, isEmpty, sortBy } from 'lodash'
 import { injectState, provideState } from 'reaclette'
+import { isSrWritable } from 'xo'
+import { Pool } from 'render-xo-item'
+import { SelectPool, SelectSr } from 'select-objects'
 
 export default decorate([
   provideState({
-    initialState: ({ multi }) => ({
-      pools: multi ? [] : undefined,
-    }),
     effects: {
       onChangePool(__, pools) {
-        const noDefaultSr = Array.isArray(pools)
-          ? pools.some(pool => pool.default_SR === undefined)
-          : pools.default_SR === undefined
-        if (noDefaultSr) {
-          error(_('hubSrErrorTitle'), _('noDefaultSr'))
-        } else {
-          this.props.onChange({
-            pools,
-            pool: pools,
-          })
-          return {
-            pools,
+        const { multi, onChange, value } = this.props
+        const _defaultSrByPool = value.mapPoolsSrs
+        if (multi) {
+          const _pool = Array.isArray(pools) ? pools : [pools]
+          for (const pool of differenceBy(_pool, value.pools, 'id')) {
```

Comment:
To avoid handling multiple cases, the modification of a value of the state should only be triggered by a user action.

To do this you can:
- put a default SR for each select, please see the example bellow:
```js
<SelectSr
  {...props}
  value={defined(mapPoolsSrs[poolId], pool.default_SR)}
>
```
- affect a default SR to undefined pools' SRs after the submission of the form

This solution will avoid calculations for each pool change and will simplify this implementation.

Reply:
Clever Idea :+1:  I tested it and it works :tada: